### ISSUE TITLE
bug: quick-lint misses `i = { a: }`

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1231]
Given the following code:

```js
let i = {a:};
```

quick-lint should report an error, but doesn't
